scripts/metadata_enrichment/tob_bioheart_sequencing_dates.py

```diff
@@ -0,0 +1,234 @@
+import asyncio
+from collections import defaultdict
+
+import pandas as pd
+
+from metamist.apis import AssayApi
+from metamist.graphql import gql, query
+from metamist.models import AssayUpsert
+
+# Extract required data from metamist
+QUERY_PROJECT_ASSAYS = gql(
+    """
+    query ProjectAssays($datasetName: String!) {
+    project(name: $datasetName) {
+        samples {
+            externalId
+            meta
+            assays {
+                id
+                meta
+            }
+        }
+        }
+    }
+    """,
+)
+
+
+def query_metamist():
+    """
+    Query metamist for bioheart and tob-wgs datasets and map all tube ids to samples
+    via call to create_default_dict.
+    Later calls append_dictionaries(), which collates dicts from the two projects.
+
+    :return: Dictionary of lists, mapping tube IDs to samples retrieved from metamist
+            for both bioheart and tob-wgs datasets
+    :rtype: defaultdict
+    """
+    # gives us json output from graphql query
+    # Results are seen in graphql interface as a dictionary
+    query_result_tob = query(QUERY_PROJECT_ASSAYS, variables={'datasetName': 'tob-wgs'})
+    query_result_bioheart = query(
+        QUERY_PROJECT_ASSAYS,
+        variables={'datasetName': 'bioheart'},
+    )
+
+    # list of dictionaries of samples per External ID
+    tob_samples = query_result_tob['project']['samples']
+    bioheart_samples = query_result_bioheart['project']['samples']
+
+    # Create default dicts for the two datasets: maps tube IDs to samples
+    tob_kccg_id = 'KCCG FluidX tube ID'
+    fluidx_to_assay_ids_tob = create_default_dict(tob_samples, tob_kccg_id)
+    bioheart_kccg_id = 'fluid_x_tube_id'
+    fluidx_to_assay_ids_bioheart = create_default_dict(
+        bioheart_samples,
+        bioheart_kccg_id,
+    )
+
+    return append_dictionaries(fluidx_to_assay_ids_tob, fluidx_to_assay_ids_bioheart)
+
+
+def create_default_dict(samples: list, kccg_id: str) -> defaultdict:
```

Comment:
I would rename this function something like "`create_fluidx_id_to_assay_ids_dict`" - or something to that effect. Current name is a bit too broad imo.

Reply:
Very good idea. Updated.